src/Psy/ColorMode.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy;
+
+/**
+ * A configuration object that represents one of 3 available color modes:
+ *   1. "auto" (the default): if color support is detected, output contains colors
+ *   2. "forced": force colors in output
+ *   3. "disabled": disable colors in output
+ */
+class ColorMode
```

Comment:
It feels like all of these are twisting ourselves into contortions to work around PHP's lack of enums and the ability to typehint against them :)

Remember that these are personal configuration options, not mission critical framework integrations. If something goes wrong, the user just has the wrong color preferences.

In keeping with the Robustness Principle, operators don't really matter. Sure, you _should_ be using the constants. But if you pass something in that `==` the value of the constant, that should be good enough for PsySH. The same way we don't throw an error if you pass something that `==` false to one of the boolean configs, we just silently convert it into what we were expecting.

Now, if you pass something that isn't `==` one of the available options, it's totally reasonable to validate it when setting and throw an exception. But we could do that with one line of code, vs the 100 line class approach.

IFF this were a strongly typed language, with expressive typehinting, and enums, and all sorts of other things, then we could use those language features to do this "right". But we still wouldn't use classes for it :P